File: src/com/fenghuo/notes/db/PreferenceHelper.java
package com.fenghuo.notes.db;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceHelper {

    private static final String KEY_PWD = "k_pwd";

    private SharedPreferences mPreferences;

    public PreferenceHelper(Context context) {
        mPreferences = context.getSharedPreferences("temp", Activity.MODE_PRIVATE);
    }

    /**
     * 读取上一次的草稿
     *
     * @return
     */
    public String getLast() {
        String str = "";
        str = mPreferences.getString("last", "");
        return str;
    }

    /**
     * 保存草稿
     *
     * @param str
     */
    public void saveLast(String str) {
        SharedPreferences.Editor editor = mPreferences.edit();
        editor.putString("last", str);
        editor.apply();
    }

    public String getPatternPwd() {
        return mPreferences.getString(KEY_PWD, null);
    }

    public void setPatternpwd(String pwd) {
        mPreferences.edit().putString(KEY_PWD, pwd).apply();
    }

    public void delPatternpwd() {
        mPreferences.edit().remove(KEY_PWD).apply();
    }
}

File: src/com/fenghuo/notes/db/PreferenceHelper.kt
package com.fenghuo.notes.db

import android.app.Activity
import android.content.Context
import android.content.SharedPreferences

class PreferenceHelper(context: Context) {

    private val mPreferences: SharedPreferences

    init {
        mPreferences = context.getSharedPreferences("temp", Activity.MODE_PRIVATE)
    }

    /**
     * 读取上一次的草稿

     * @return
     */
    val last: String
        get() {
            return mPreferences.getString("last", "")
        }

    /**
     * 保存草稿

     * @param str
     */
    fun saveLast(str: String) {
        val editor = mPreferences.edit()
        editor.putString("last", str)
        editor.apply()
    }

    fun removeLastDraft() {
        val editor = mPreferences.edit()
        editor.remove("last");
        editor.apply()
    }

    fun saveLastWithID(id: Long, str: String) {
        val editor = mPreferences.edit()
        editor.putString("last_draft_with_id:$id", str);
        editor.apply()
    }

    fun getLastDraftById(id: Long): String? {
        return mPreferences.getString("last_draft_with_id:$id", null);
    }

    fun removeLastDraft(id: Long) {
        val editor = mPreferences.edit()
        editor.remove("last_draft_with_id:$id");
        editor.apply()
    }

    val patternPwd: String?
        get() = mPreferences.getString(KEY_PWD, null)

    fun setPatternpwd(pwd: String) {
        mPreferences.edit().putString(KEY_PWD, pwd).apply()
    }

    fun delPatternpwd() {
        mPreferences.edit().remove(KEY_PWD).apply()
    }

    companion object {

        private val KEY_PWD = "k_pwd"
    }
}
